engine/Shopware/Core/sBasket.php

```
@@ -421,7 +421,7 @@ public function sInsertDiscount()
             $tax = $this->config->get('sDISCOUNTTAX');
         }
 
-        if (!$tax) {
+        if (!$tax && $tax != 0) {
```

```suggestion
        if ($tax === false) {
```

---

Hi @aragon999 ,
this way looks much nicer, I wasn't sure, if there is another datatype that can trigger the fallback.

---

As far as I can tell, the only case which needs to be handled is that `getMaxTax()` returns false.

It would still change the logic slightly, as when previously when the tax `0` has been configured in `sDISCOUNTTAX` the tax wont be `19` anymore. But that is I think in all cases like this.

---

I would also prefer a type safe comparison. but then we need to max sure, that `$tax` is really a `float` or `false` at this point. 

additionally to that, I would like to remove the fallback to a hardcoded value :see_no_evil: 

what do you think? throwing an exception or getting a tax value from somewhere else?